src/main/java/org/icgc/argo/program_service/services/ProgramServiceFacade.java

```diff
@@ -173,13 +173,34 @@ public UpdateProgramResponse updateProgram(UpdateProgramRequest request) {
         programService.updateProgram(
             programToUpdate,
             updatingProgram,
+            null,
             program.getCancerTypesList(),
             program.getPrimarySitesList(),
             program.getInstitutionsList(),
             program.getCountriesList());
     return programConverter.programEntityToUpdateProgramResponse(updatedProgram);
   }
 
+  @Transactional
+  public UpdateProgramResponse updateProgramWithDataCenter (UpdateProgramRequest request, DataCenterDetailsDTO dataCenterDetailsDTO) {

```

Comment:
Can this new function just take the UpdateProgramDTO instead of relying on the GRPC message type? We want to eventually remove all uses of the GRPC code.